url.bs

```diff
@@ -272,6 +272,94 @@ for further processing.
 U+0020 SPACE, U+0023 (#), U+0025 (%), U+002F (/), U+003A (:), U+003F (?), U+0040 (@), U+005B ([),
 U+005C (\), or U+005D (]).
 
+<p>A <a for=/>host</a>'s <dfn for=host export>public suffix</dfn> is the portion of a
+<a for=/>host</a> which is controlled by a registrar, public or otherwise. To obtain
+<var>host</var>'s <a for=host>public suffix</a>, run these steps:
+
+<ol>
+ <li><p>If <var>host</var> is not a <a>domain</a>, then return null.
+
+ <li><p>Return the <a for=host>public suffix</a> obtained by executing the
+ <a href="https://publicsuffix.org/list/">algorithm</a> defined by the Public Suffix List on
+ <var>host</var>. [[!PSL]].
+</ol>
+
+<p>A <a for=/>host</a>'s <dfn for=host export>registrable domain</dfn> is a <a>domain</a> that could
+be registered at a registry. To obtain <var>host</var>'s <a for=host>registrable domain</a>, run
```

Comment:
"is its public suffix including one domain label preceding its public suffix". Again, boring, but factual?

Reply:
So a given host may have multiple public suffixes expressed within it.

Perhaps:
The domain formed by the most specific public suffix for host, along with the domain label immediately preceding it?

From a spec question, what do you want this definition to entail for the `appspot.com` case?

That is,
- `foo.bar.appspot.com` is "obviously" going to return `bar.appspot.com` as the registerable domain (with `appspot.com` as the public suffix), and the same would be expected if just `bar.appspot.com`.
- What do you expect this machinery to return for `appspot.com`? `appspot.com` is on the PSL, so that is a public suffix, but `appspot.com` is also a registerable domain under the `com` PSL.

I seem to recall that different platform features interpret that differently (navigation vs cookies, for example)

Reply:
@mikewest wrote that the registrable domain would be null in such a case (we have github.io as example).

Reply:
I’d need to reaudit the Chrome code to figure out which cases are web visible. The results differ in this case based on whether or not you include private suffices and whether you treat wildcards as implicit entries of the parent. Chrome and FF differ on the latter, and the former is specified by the caller.